pandas/tests/groupby/test_groupby.py

```diff
@@ -1752,3 +1752,81 @@ def test_groupby_groups_in_BaseGrouper():
     result = df.groupby(['beta', pd.Grouper(level='alpha')])
     expected = df.groupby(['beta', 'alpha'])
     assert(result.groups == expected.groups)
+
+
+def test_groupby_indices():
+    # GH 26860
+    # Test if DataFrame Groupby builds gb.indices correctly.
+
+    int_series = pd.Series([1, 2, 3])
+    int_series_cat = int_series.astype('category')
+    float_series = pd.Series([1., 2., 3.])
+    float_series_cat = float_series.astype('category')
+    dt_series = pd.to_datetime(['2018Q1', '2018Q2', '2018Q3'])
+    dt_series_cat = dt_series.astype('category')
+    period_series = dt_series.to_period('Q')
+    period_series_cat = period_series.astype('category')
+
+    df = pd.DataFrame({
+        'int_series': int_series,
+        'int_series_cat': int_series_cat,
+        'float_series': float_series,
+        'float_series_cat': float_series_cat,
+        'dt_series': dt_series,
+        'dt_series_cat': dt_series_cat,
+        'period_series': period_series,
+        'period_series_cat': period_series_cat
+    })
+    col_order = [
+        'int_series',
+        'int_series_cat',
+        'float_series',
+        'float_series_cat',
+        'dt_series',
+        'dt_series_cat',
+        'period_series',
+        'period_series_cat'
+    ]
+    df = df[col_order]
+    from itertools import combinations
+
+    dts = [
+        np.datetime64('2018-01-01T00:00:00.000000000'),
+        np.datetime64('2018-04-01T00:00:00.000000000'),
+        np.datetime64('2018-07-01T00:00:00.000000000')
+    ]
+    pers = [pd.Period(dt, freq='Q') for dt in dts]
+
+    target_key_choices = [
+        [1, 1, 1.0, 1.0, dts[0], dts[0], pers[0], pers[0]],
+        [2, 2, 2.0, 2.0, dts[1], dts[1], pers[1], pers[1]],
+        [3, 3, 3.0, 3.0, dts[2], dts[2], pers[2], pers[2]]
+    ]
+    target_indices_values = [
+        np.array([i])
+        for i in range(df.shape[0])
+    ]
+    n_choices = len(df.columns)
+
+    for n in range(1, n_choices + 1):
```

Comment:
you can parameterize this function  rather than doing this as it makes the test very opaque